src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -450,12 +450,10 @@ private static IEnumerable<TargetFrameworkInformation> GetTargetFrameworkInforma
                     restoreType == ProjectStyle.DotnetToolReference)
                 {
                     var packageTargetFallback = MSBuildStringUtility.Split(item.GetProperty("PackageTargetFallback"))
-                        .Select(NuGetFramework.Parse)
-                        .ToList();
+                        .Select(NuGetFramework.Parse);
```

Comment:
This gets enumerated multiple times, so this might not be reducing allocations, it'd be adding more because we'd end up running at least double the amount of `NuGetFramework.Parse` calls.